File: energy_settings/src/main/java/de/cyface/energy_settings/EnergySettingDialog.java
/*
 * Copyright 2019 Cyface GmbH
 *
 * This file is part of the Cyface SDK for Android.
 *
 * The Cyface SDK for Android is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * The Cyface SDK for Android is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with the Cyface SDK for Android. If not, see <http://www.gnu.org/licenses/>.
 */
package de.cyface.energy_settings;

import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentManager;

/**
 * This class is used to identify dialogs which are opened by this library.
 * <p>
 * This is needed to only close such dialogs in {@link TrackingSettings#dismissAllDialogs(FragmentManager)} and to not
 * interfere with other dialogs.
 *
 * @author Armin Schnabel
 * @version 1.0.0
 * @since 1.1.0
 */
abstract class EnergySettingDialog extends DialogFragment {
    // Nothing to do
}

File: energy_settings/src/main/kotlin/de/cyface/energy_settings/EnergySettingDialog.kt
/*
 * Copyright 2019-2025 Cyface GmbH
 *
 * This file is part of the Cyface Energy Settings for Android.
 *
 * The Cyface Energy Settings for Android is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * The Cyface Energy Settings for Android is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with the Cyface Energy Settings for Android. If not, see <http://www.gnu.org/licenses/>.
 */
package de.cyface.energy_settings

import android.content.Context
import android.content.Intent
import androidx.fragment.app.DialogFragment
import com.afollestad.materialdialogs.DialogBehavior
import com.afollestad.materialdialogs.MaterialDialog

/**
 * This class contains shared static methods of subclasses.
 *
 * It's also needed to only close such dialogs in [TrackingSettings.dismissAllDialogs] and to not interfere
 * with other dialogs. This is only used in the Fragment based Dialog implementations, not the MaterialDialog based.
 *
 * @author Armin Schnabel
 * @version 2.0.1
 * @since 1.1.0
 */
internal abstract class EnergySettingDialog : DialogFragment() {
  companion object {
    /**
     * For all instances of [MaterialDialog].
     */
    internal var DIALOG_BEHAVIOUR: DialogBehavior = MaterialDialog.DEFAULT_BEHAVIOR

    /**
     * The resource pointing to the text used as template for the feedback email in [intent].
     */
    private val feedbackErrorDescriptionRes = R.string.feedback_error_description
    /**
     * The resource pointing to the text used to ask the use to choose an email app which the [intent] should open.
     */
    internal val chooseEmailAppRes = R.string.feedback_choose_email_app

    /**
     * The Android `Intent` to be started when the positive button was clicked (opens feedback email).
     */
    fun intent(context: Context, recipientEmail: String): Intent {
      return TrackingSettings.generateFeedbackEmailIntent(
        context,
        context.getString(feedbackErrorDescriptionRes),
        recipientEmail
      )
    }
  }
}
